Sources/Extensions/SwiftStdlib/ArrayExtensions.swift

```diff
@@ -231,6 +231,23 @@ public extension Array where Element: Equatable {
 		return indices
 	}
 
+    @discardableResult
+    /// SwifterSwift: Remove first instance of an element from array.
+    ///
+    ///        [1, 2, 2, 3, 4, 2, 5].removeElement(2) -> [1, 2, 3, 4, 2, 5]
+    ///        ["h", "e", "l", "l", "o"].removeElement("e") -> ["h", "l", "l", "o"]
+    ///
+    /// - Parameter element: element to remove.
+    /// - Returns: the removed element, otherwise nil.
+    public mutating func removeElement(_ element: Element) -> Element? {
```

Comment:
This doesn't give any indication to the user that only the first element is removed.
Maybe `removeFirst(where condition: (Element) -> Bool) -> Element?`

Also, this can't be done by the Swift Standard Library? 🤔

Reply:
`removeElement` removes only the specified element. Why should it be puzzling for the user?

Reply:
Btw this can’t be done by Swift lib. 
You can remove based on the `index` or by using `filter`

Reply:
> removeElement removes only the specified element. Why should it be puzzling for the user?

Because the name doesn't convey to the user that **we're only removing the first element, not all elements**

Reply:
So I'm sure that I added this comment already, but it seems to be gone. I agree that the name isn't perfectly clear. How about `removeFirst(element: Element)`? You can't use `removeFirst(_ element: Element)` because then it would be ambiguous, if `Element == Int`, with `removeFirst(_ n: Int)`.

Reply:
Needs to be throwing if we do `removeFirst(where:)`